subprojects/testing-native/src/integTest/groovy/org/gradle/nativeplatform/test/xctest/XCTestTestFrameworkIntegrationTest.groovy

```diff
@@ -16,22 +16,43 @@
 
 package org.gradle.nativeplatform.test.xctest
 
-import org.gradle.nativeplatform.fixtures.app.SwiftXCTestTestFrameworkBundle
+import org.gradle.integtests.fixtures.SourceFile
+import org.gradle.nativeplatform.fixtures.app.XCTestCaseElement
+import org.gradle.nativeplatform.fixtures.app.XCTestSourceElement
+import org.gradle.nativeplatform.fixtures.app.XCTestSourceFileElement
 import org.gradle.testing.AbstractTestFrameworkIntegrationTest
 import org.gradle.util.Requires
 import org.gradle.util.TestPrecondition
 
 @Requires([TestPrecondition.SWIFT_SUPPORT, TestPrecondition.MAC_OS_X])
 class XCTestTestFrameworkIntegrationTest extends AbstractTestFrameworkIntegrationTest {
+    def setup() {
+        settingsFile << "rootProject.name = 'app'"
+        buildFile << """
+            apply plugin: 'xctest'
+        """
+    }
+
     @Override
     void createPassingFailingTest() {
         def testBundle = new SwiftXCTestTestFrameworkBundle()
         testBundle.writeToProject(testDirectory)
 
-        settingsFile << "rootProject.name = 'app'"
-        buildFile << """
-    apply plugin: 'xctest'
-"""
+    }
+
+    @Override
+    void createEmptyProject() {
+        file("src/test/swift/NoTests.swift") << """
+            func someSwiftCode() {
+            }
+        """
+    }
+
+    @Override
+    void changeTests() {
```

Comment:
Ditto. I'm wondering if there is any way to generalize this method and `craeteEmptyProject` so that we don't have so much copy pasta.